openedx/license-manager
# feat: adding management command to check number of distinct catalog q…

…ueries used by SubscriptionPlan catalogs

**WARNING**: Please run `make lint` locally, as this repository is currently blocked from running pylint
automatically via Github Actions.

## Description

* Adds `get_distinct_catalog_queries` method to the `EnterpriseCatalogApiClient`
* Adds `validate_num_catalog_queries` management command to check number of distinct `CatalogQuery` objects used by `EnterpriseCatalog` objects associated with `SubscriptionPlans` that aren't FIUO. 

## Ticket Link

[ENT-4039](https://openedx.atlassian.net/browse/ENT-4039)

## Testing Instructions

* Use the app shell to run the management command. Use the worker logs to observe the results which will differ based on the devstack data of the tester. 
* Verify that, based on the catalogs used by your devstack SubscriptionPlans (and the CatalogQueries associated with them), the correct result is logged. 
  * 3 distinct CatalogQueries --> Success
  * Any other number of distinct CatalogQueries --> Error

## Post-review

Squash commits into discrete sets of changes
